File: fsharp/raindrops/Raindrops.fs
module Raindrops

let convert (number: int): string =
    [ 3, "Pling"; 5, "Plang"; 7, "Plong" ]
    |> List.choose (fun (n, pl) -> if number % n = 0 then Some pl else None)
    |> function
        | [] -> number.ToString()
        | plings -> String.concat "" plings

File: lua/raindrops/raindrops.lua
return function(n)
    local function isDivisibleBy(m) return n % m == 0 end
    local result = ''
    if isDivisibleBy(7) then result = 'Plong' .. result end
    if isDivisibleBy(5) then result = 'Plang' .. result end
    if isDivisibleBy(3) then result = 'Pling' .. result end
    if result == '' then result = tostring(n) .. result end
    return result
end
